Class availability page

willgearty opened the pull request:
This implements an availability page for a class. It looks just like the teacher/volunteer availability pages, but shows whether the teachers for a class are available or not for the time blocks:
![image](https://user-images.githubusercontent.com/7232514/56084214-b01b0c00-5de4-11e9-9a41-76b1ea319287.png)

If there are teachers that aren't available for a particular time block, hovering over that time block will show which teachers are already teaching:
![image](https://user-images.githubusercontent.com/7232514/56084219-c2954580-5de4-11e9-9a45-98ab14a741ee.png)

and/or which teachers are not available:
![image](https://user-images.githubusercontent.com/7232514/56084221-c88b2680-5de4-11e9-873e-61d882a9e314.png)

If a section of the class is scheduled, hovering over that time block will show that:
![image](https://user-images.githubusercontent.com/7232514/55693115-14ac1600-5962-11e9-9d16-184299ff77d1.png)

I added links to this page from the `manageclass_class_info.html` template, which is included in the `/manageclass` and `/classsearch` pages. I also included a link to this page from the manage coteachers page. Finally, I included links in the scheduler, so you can go directly to this page while scheduling classes. While I was at it, I added return links to the class manage page from the manage coteachers and class availability pages.

Fixes #2638 and #1402.

willgearty commented:
This pull request **introduces 1 alert** when merging 4458379c9ab3efee3cdd519ec4b68c98c609271b into 33a71d682523ddb24c9b5d15578b4d527eddda32 - [view on LGTM.com](https://lgtm.com/projects/g/learning-unlimited/ESP-Website/rev/pr-2b522aacae67de615f2327fb11c9e43e52d4a030)

**new alerts:**

* 1 for Unused local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

kkbrum commented:
I like this a lot. One thought I have is that in my mind, "unavailable" and "teaching something else" are kind of different because the approaches to fixing them are different. Do you think it might make sense to specify that someone is teaching some other class as opposed to calling them unavailable if that is the case?

kkbrum commented:
On the class availability page, would we want to note somewhere the length of the class? Just so we know if it's a 2 block class, that we are looking for 2 consecutive blocks of availability? just for easy reference

willgearty commented:
Thanks for the great ideas @kkbrum! The hover details now distinguish between whether a teacher is already teaching or just unavailable. I've also added the length of the class to the page. I've updated the images in the OP to reflect these changes.

kkbrum commented:
Could we put a line break before teaching here?

![image](https://user-images.githubusercontent.com/25392075/56250729-9694f100-607e-11e9-8ace-ebd3a5c1258c.png)

willgearty commented:
@kkbrum Let me know if that works.
